src/app/public/src/modules/clipboard/clipboard.service.ts

```diff
@@ -0,0 +1,79 @@
+// adapted from ngx-clipboard
+
+import { Injectable, Renderer2, RendererFactory2 } from '@angular/core';
+import { StacheWindowRef } from '../shared';
+
+@Injectable()
+export class StacheCopyToClipboardService {
+  private copyTextArea: HTMLTextAreaElement;
+  private document: any;
+  private renderer: Renderer2;
+
+  constructor(
+    private windowRef: StacheWindowRef,
+    private rendererFactory: RendererFactory2
+  ) {
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+    this.document = this.windowRef.nativeWindow.document;
+  }
+
+  public copyContent(element: HTMLElement) {
+    let copyTarget = this.getValidInputElement(element);
+    this.copyFromInputElement(copyTarget);
+  }
+
+  public verifyCopyCommandEnabled() {
+    return this.document.execCommand('copy');
```

Comment:
We should use `queryCommandSupported`, if you load a page before any user interaction, execCommand returns false.